airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/write/BasicPerformanceTest.kt

```diff
@@ -0,0 +1,344 @@
+import io.airbyte.cdk.load.data.IntegerType
+import io.airbyte.cdk.load.data.ObjectTypeWithoutSchema
+import io.airbyte.cdk.load.data.StringType
+import io.airbyte.cdk.load.data.TimeTypeWithTimezone
+import io.airbyte.cdk.load.data.TimeTypeWithoutTimezone
+import io.airbyte.cdk.load.data.TimestampTypeWithTimezone
+import io.airbyte.cdk.load.data.TimestampTypeWithoutTimezone
+import io.airbyte.cdk.load.message.DestinationRecordStreamComplete
+import io.airbyte.cdk.load.test.util.ConfigurationUpdater
+import io.airbyte.cdk.load.test.util.FakeConfigurationUpdater
+import io.airbyte.cdk.load.test.util.IntegrationTest
+import io.airbyte.cdk.load.test.util.destination_process.DestinationProcess
+import io.airbyte.cdk.load.test.util.destination_process.DestinationProcessFactory
+import io.airbyte.protocol.models.Jsons
+import io.github.oshai.kotlinlogging.KotlinLogging
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.LocalTime
+import java.time.OffsetDateTime
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
+import kotlin.time.measureTime
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.apache.commons.lang3.RandomStringUtils
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInfo
+import org.junit.jupiter.api.extension.ExtendWith
+import uk.org.webcompere.systemstubs.environment.EnvironmentVariables
+import uk.org.webcompere.systemstubs.jupiter.SystemStub
+import uk.org.webcompere.systemstubs.jupiter.SystemStubsExtension
+
+private val log = KotlinLogging.logger {}
+
+data class NamedField(val name: String, val type: AirbyteType, val sample: Any)
+
+/** Defines a performance test scenario. */
+interface PerformanceTestScenario {
+    data class Summary(
+        val records: Long,
+        val size: Long,
+        val expectedRecordsCount: Long,
+    )
+
+    /** The catalog used for the performance test. */
+    val catalog: DestinationCatalog
+
+    /**
+     * The main method from the performance scenario.
+     *
+     * This would be where records are emitted to the destination. How, is up to the scenario to
+     * define.
+     */
+    fun send(destination: DestinationProcess)
+
+    /**
+     * Returns the expectations from the test scenario: how many records were emitted, how many
+     * records are expected to be written in the final table (in the case of duplicates, this should
+     * be the number of distinct records) and the volume of data emitted.
+     */
+    fun getSummary(): Summary
+}
+
+/** Interface to implement for destination that support data validation. */
+interface DataValidator {
+    /**
+     * Returns the count of how many records are present for the stream in the final table. null if
+     * not found.
+     */
+    fun count(spec: ConfigurationSpecification, stream: DestinationStream): Long?
```

Comment:
maybe worth putting this method on DataDumper directly? then we get a default impl for free, even if it's really dumb
```kotlin
interface DataDumper {
  fun count(...): Long? = dumpRecords(spec, stream).size()
}
```